tedana/metrics/dependence.py

```diff
@@ -0,0 +1,604 @@
+    F_S0_maps = np.zeros([n_voxels, n_components])
+
+    for i_comp in range(n_components):
+        # size of comp_betas is (n_echoes, n_samples)
+        comp_betas = np.atleast_3d(me_betas)[:, :, i_comp].T
+        alpha = (np.abs(comp_betas) ** 2).sum(axis=0)
+
+        # Only analyze good echoes at each voxel
+        for j_echo in np.unique(adaptive_mask[adaptive_mask >= 3]):
+            mask_idx = adaptive_mask == j_echo
+            alpha = (np.abs(comp_betas[:j_echo]) ** 2).sum(axis=0)
+
+            # S0 Model
+            # (S,) model coefficient map
+            coeffs_S0 = (comp_betas[:j_echo] * X1[:j_echo, :]).sum(axis=0) / (
+                X1[:j_echo, :] ** 2
+            ).sum(axis=0)
+            pred_S0 = X1[:j_echo, :] * np.tile(coeffs_S0, (j_echo, 1))
+            SSE_S0 = (comp_betas[:j_echo] - pred_S0) ** 2
+            SSE_S0 = SSE_S0.sum(axis=0)  # (S,) prediction error map
+            F_S0 = (alpha - SSE_S0) * (j_echo - 1) / (SSE_S0)
+            F_S0[F_S0 > f_max] = f_max
+            F_S0_maps[mask_idx, i_comp] = F_S0[mask_idx]
+
+            # T2 Model
+            coeffs_T2 = (comp_betas[:j_echo] * X2[:j_echo, :]).sum(axis=0) / (
+                X2[:j_echo, :] ** 2
+            ).sum(axis=0)
+            pred_T2 = X2[:j_echo] * np.tile(coeffs_T2, (j_echo, 1))
+            SSE_T2 = (comp_betas[:j_echo] - pred_T2) ** 2
+            SSE_T2 = SSE_T2.sum(axis=0)
+            F_T2 = (alpha - SSE_T2) * (j_echo - 1) / (SSE_T2)
+            F_T2[F_T2 > f_max] = f_max
+            F_T2_maps[mask_idx, i_comp] = F_T2[mask_idx]
+
+    return F_T2_maps, F_S0_maps
+
+
+def threshold_map(maps, mask, ref_img, threshold, csize=None):
+    """Perform cluster-extent thresholding.
+
+    Parameters
+    ----------
+    maps : (M x C) array_like
+        Statistical maps to be thresholded.
+    mask : (S) array_like
+        Binary mask.
+    ref_img : img_like
+        Reference image to convert to niimgs with.
+    threshold : :obj:`float`
+        Value threshold to apply to maps.
+    csize : :obj:`int` or :obj:`None`, optional
+        Minimum cluster size. If None, standard thresholding (non-cluster-extent) will be done.
+        Default is None.
+
+    Returns
+    -------
+    maps_thresh : (M x C) array_like
+    """
+    n_voxels, n_components = maps.shape
+    maps_thresh = np.zeros([n_voxels, n_components], bool)
+    if csize is None:
+        csize = np.max([int(n_voxels * 0.0005) + 5, 20])
+    else:
+        csize = int(csize)
+
+    for i_comp in range(n_components):
+        # Cluster-extent threshold and binarize F-maps
+        ccimg = io.new_nii_like(
+            ref_img, np.squeeze(utils.unmask(maps[:, i_comp], mask))
+        )
+
+        maps_thresh[:, i_comp] = utils.threshold_map(
+            ccimg, min_cluster_size=csize, threshold=threshold, mask=mask, binarize=True
+        )
+    return maps_thresh
+
+
+def threshold_to_match(maps, n_sig_voxels, mask, ref_img, csize=None):
```

Comment:
@jbteves- `dice_FS0` and `dice_FT2` rely on cluster-extent thresholded maps made using this function, wherein the F-statistic maps for S0 or T2* models are cluster-extent thresholded to have a total number of significant voxels that matches (as closely as possible) the number in the cluster-extent thresholded Z-statistic map. I think that could cause some of the differences, since `tedana` doesn't currently try to match the number of significant voxels very well.

Reply:
The inputs to the DICE measures are used in other measures. The big thing that might make the DICE measures change is what is or isn't masked in the volume. signal-noise_t also relies heavily on the volume mask. Maybe the mask has changed between the two versions?
I wouldn't worry about kappa_ratio. That's a measure that's calculated between components within the decision tree, so, anything that affects whether a component is rejected may then affect kappa_ratio. If we fix the other issues, I suspect kappa_ratio (and d_table) will also be fixed.

Reply:
>The inputs to the DICE measures are used in other measures. The big thing that might make the DICE measures change is what is or isn't masked in the volume.

Agreed. That is definitely causing differences here.

> signal-noise_t also relies heavily on the volume mask. Maybe the mask has changed between the two versions?

It's possible, but I can't see where that would happen. `signal-noise_t` relies on the thresholded z-statistic map, which is thresholded using the normal procedure. The F-T2 and F-S0 thresholded maps are the ones that should be affected by the changes in this PR.